packages/wallet/backend/src/account/service.ts

```diff
@@ -295,4 +308,84 @@ export class AccountService implements IAccountService {
 
     return account
   }
+
+  public async createExchangeQuote({
+    userId,
+    accountId,
+    assetCode,
+    amount
+  }: CreateExchangeQuote): Promise<QuoteWithFees> {
+    const accountFrom = await Account.query()
+      .findById(accountId)
+      .withGraphFetched({ paymentPointers: true })
+      .modifyGraph('paymentPointers', (builder) => {
+        builder.where({ active: true }).orderBy('createdAt', 'ASC').limit(1)
+      })
+
+    if (!accountFrom) {
+      throw new NotFound(`The source account does not exist.`)
+    }
+
+    const assetInRafiki = (
```

Comment:
please extract the asset validation in a separate method

Reply:
as this may be needed in different services, probably the best place to be is in rafiki service